lib/ansiblelint/utils.py

```diff
@@ -599,55 +599,109 @@ def get_first_cmd_arg(task):
+    return yaml_skip
 
+
+def _append_skipped_rules(pyyaml_data, file_text, file_type):
+    # parse file text using 2nd parser library
     yaml = ruamel.yaml.YAML()
     ruamel_data = yaml.load(file_text)
 
+    # if file type is 'meta', add skips to metadata block and return
+    if file_type == 'meta':
+        skipped_rules = _get_rule_skips_from_yaml(ruamel_data)
+        if skipped_rules:
+            pyyaml_data[0]['skipped_rules'] = skipped_rules
+        return pyyaml_data
+
+    # create list of blocks of tasks or nested tasks
     if file_type in ('tasks', 'handlers'):
-        ruamel_tasks = ruamel_data
-        pyyaml_tasks = pyyaml_data
+        ruamel_task_blocks = ruamel_data
+        pyyaml_task_blocks = pyyaml_data
     elif file_type == 'playbook':
         try:
-            ruamel_tasks = []
-            pyyaml_tasks = []
-            for ruamel_play, pyyaml_play in zip(ruamel_data, pyyaml_data):
-                for key in PLAYBOOK_TASK_KEYWORDS:
-                    ruamel_tasks.extend(ruamel_play.get(key, []))
-                    pyyaml_tasks.extend(pyyaml_play.get(key, []))
+            pyyaml_task_blocks = _get_task_blocks_from_playbook(pyyaml_data)
+            ruamel_task_blocks = _get_task_blocks_from_playbook(ruamel_data)
         except (AttributeError, TypeError):
+            # TODO(awcrosby): running ansible-lint on any .yml file will
+            # assume it is a playbook, check needs to be added higher in the
+            # call stack, and can remove this except
             return pyyaml_data
-    elif file_type == 'meta':
-        if not isinstance(pyyaml_data, list):
-            return pyyaml_data
-        ruamel_tasks = [ruamel_data]
-        pyyaml_tasks = pyyaml_data
     else:
-        return pyyaml_data
+        raise RuntimeError('Unexpected file type')
 
-    if len(ruamel_tasks) != len(pyyaml_tasks):
-        return pyyaml_data
+    # get tasks from blocks of tasks
+    pyyaml_tasks = _get_tasks_from_blocks(pyyaml_task_blocks)
+    ruamel_tasks = _get_tasks_from_blocks(ruamel_task_blocks)
 
+    # append skipped_rules for each task
     for ruamel_task, pyyaml_task in zip(ruamel_tasks, pyyaml_tasks):
-        skipped_rules = _get_rule_skips_from_task(ruamel_task)
+        if pyyaml_task.get('name') != ruamel_task.get('name'):
+            raise RuntimeError('Error in matching skip comment to a task')
+        skipped_rules = _get_rule_skips_from_yaml(ruamel_task)
         if skipped_rules:
             pyyaml_task['skipped_rules'] = skipped_rules
 
     return pyyaml_data
 
 
-def _get_rule_skips_from_task(task):
+def _get_task_blocks_from_playbook(playbook):
+    """Return parts of playbook that contains tasks, and nested tasks."""
+    PLAYBOOK_TASK_KEYWORDS = [
+        'tasks',
+        'pre_tasks',
+        'post_tasks',
+        'handlers',
+    ]
+
+    task_blocks = []
+    for play in playbook:
+        for key in PLAYBOOK_TASK_KEYWORDS:
+            task_blocks.extend(play.get(key, []))
```

Comment:
I'll be simpler if you turn this into a generator
```suggestion
            yield play.get(key, [])
```
And than `list()` its call if needed.

Reply:
(use `itertools.product()` regardless of whether you choose to use a generator or a comprehension)